### PR TITLE
[iTranslate]: add new extension iTranslate

### PR DESCRIPTION
**Pull request by 404J**
## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by 404J** [comment]
Help! does someone review this PR?

**Comment by danielsequeira** [comment]
Hi @404J,

Thanks for you contribution.

The extension that you built seems to do exactly the same as [Easy Dictionary](https://www.raycast.com/isfeng/easydict), which includes all the same sources as yours. Did you know about or tried it out?

**Comment by 404J** [comment]
Hi @danielsequeira,
Before coding this extension, I had tried some translation tools like Easy Dictionary in Raycast Store, where we both offer basic word and text translation features. But I tried to differentiate myself from them, iTransate offers the ability to quickly switch the target language and saving translation results for the word to a snippet. I would also like to learn how to write a raycast extension by making the iTranslate. More importantly, I wanted to give users more options.

File: extensions/itranslate/package.json [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+      ]
+    }
+  ],
+  "preferences": [
+    {
+      "name": "langFirst",
+      "type": "dropdown",
+      "required": true,
+      "title": "First Language",
+      "description": "Select you first language",
```
Comment (danielsequeira) [inline extensions/itranslate/package.json:36]:
```suggestion
      "description": "Select your first language",
```

File: extensions/itranslate/package.json [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+          "value": "vi"
+        }
+      ]
+    },
+    {
+      "name": "langSecond",
+      "type": "dropdown",
+      "required": true,
+      "title": "Second Language",
+      "description": "Select you second language",
```
Comment (danielsequeira) [inline extensions/itranslate/package.json:93]:
```suggestion
      "description": "Select your second language",
```

File: extensions/itranslate/package.json [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+      "required": false,
+      "label": "Disable Baidu",
+      "description": "Disable Baidu translation service provider",
+      "default": false
+    },
+    {
+      "name": "tencentAppId",
+      "type": "password",
+      "required": false,
+      "title": "Tencent SecretId",
```
Comment (danielsequeira) [inline extensions/itranslate/package.json:250]:
```suggestion
      "title": "Tencent Secret ID",
```

File: extensions/itranslate/package.json [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+      "description": "Tencent SecretId",
```
Comment (danielsequeira) [inline extensions/itranslate/package.json:251]:
```suggestion
      "description": "Tencent Secret ID",
```

File: extensions/itranslate/package.json [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+      "placeholder": "Enter Tencent SecretId"
```
Comment (danielsequeira) [inline extensions/itranslate/package.json:252]:
```suggestion
      "placeholder": "Enter Tencent Secret ID"
```

File: extensions/itranslate/package.json [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+    },
+    {
+      "name": "tencentAppKey",
+      "type": "password",
+      "required": false,
+      "title": "Tencent SecretKey",
```
Comment (danielsequeira) [inline extensions/itranslate/package.json:258]:
```suggestion
      "title": "Tencent Secret Key",
```

File: extensions/itranslate/package.json [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+      "description": "Tencent SecretKey",
```
Comment (danielsequeira) [inline extensions/itranslate/package.json:259]:
```suggestion
      "description": "Tencent Secret Key",
```

File: extensions/itranslate/package.json [hunk 8]
```diff
@@ -0,0 +1,317 @@
...
+      "placeholder": "Enter Tencent SecretKey"
```
Comment (danielsequeira) [inline extensions/itranslate/package.json:260]:
```suggestion
      "placeholder": "Enter Tencent Secret Key"
```

File: extensions/itranslate/CHANGELOG.md [hunk 9]
```diff
@@ -0,0 +1,5 @@
+# iTranslate Changelog
+
+## [Initial Version] - 2022-07-30
```
Comment (danielsequeira) [inline extensions/itranslate/CHANGELOG.md:3]:
```suggestion
## [Initial Version] - 2022-08-09
```

**Comment by danielsequeira** [comment]
@404J fair enough, I just went through the code and tested the extension and left a few comments, mostly typos and nitpicking for keeping it consistent

**Comment by 404J** [comment]
@danielsequeira Thinks, I pushed a new commit and resolved these typos, and I double-checked for spelling problems elsewhere in the project

**Comment by danielsequeira** [comment]
@404J thank you so much.

I just noticed one small thing, your icon for the extension is not really 512x512, because it has some margin on it and the icon doesn't occupy the entire image, could you fix this? It makes it look smaller than other icons in Raycast

**Review by danielsequeira — APPROVED** [review]

**Comment by 404J** [comment]
@danielsequeira The size of these icons is indeed a bit problematic, I have fixed it

**Comment by 404J** [comment]
@danielsequeira wow! merged! thinks!

**Pull request merged**